container.go

```diff
@@ -56,40 +56,52 @@ func containerFromRecord(client *Client, c containers.Container) *container {
 var _ = (Container)(&container{})
 
 type container struct {
-	mu sync.Mutex
+	mu sync.RWMutex
+	id string
 
 	client *Client
 	c      containers.Container
 }
 
 // ID returns the container's unique id
 func (c *container) ID() string {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
 	return c.c.ID
 }
 
 func (c *container) Info() containers.Container {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
 	return c.c
 }
 
+func (c *container) Extensions() map[string]prototypes.Any {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+	return c.c.Extensions
```

Comment:
We should probably update the container here, same as labels below since those can be changed.

Reply:
we will have to update these to return an error, do you think that is ok?

Reply:
It's what we have for `Labels`. And probably what we should have for `Specs` since it can be updated too. Would make it consistent